lib/iris/tests/unit/cube/test_Cube.py

```diff
@@ -336,6 +336,54 @@ def test_non_lazy_aggregator(self):
         self.assertArrayEqual(result.data, np.mean(self.data, axis=1))
 
 
+class Test_collapsed__multidim_weighted(tests.IrisTest):
+    def setUp(self):
+        self.data = np.arange(6.0).reshape((2, 3))
+        self.lazydata = as_lazy_data(self.data)
+        cube_real = Cube(self.data)
+        for i_dim, name in enumerate(("y", "x")):
+            npts = cube_real.shape[i_dim]
+            coord = DimCoord(np.arange(npts), long_name=name)
+            cube_real.add_dim_coord(coord, i_dim)
+        self.cube_real = cube_real
+        self.cube_lazy = cube_real.copy(data=self.lazydata)
+        self.y_weights = np.array([0.3, 0.5])
```

Comment:
I think it might be worthwhile also having a test for `self.x_weights`, especially since non-lazy aggregation involves some ammount of transposition. Given that aggregation works for "y", infering from the code that it also works for "x" doesn't seem trivial to me.